pkg/cloudprovider/aws/instance.go

```diff
@@ -143,6 +205,13 @@ func (p *InstanceProvider) launchInstances(ctx context.Context, constraints *v1a
 	if err != nil {
 		return nil, fmt.Errorf("creating fleet %w", err)
 	}
+	insufficientCapacityOfferings := map[string]sets.String{}
+	for _, err := range createFleetOutput.Errors {
+		if InsufficientCapacityErrorCode == aws.StringValue(err.ErrorCode) {
+			createOrAppendToMapValue(insufficientCapacityOfferings, aws.StringValue(err.LaunchTemplateAndOverrides.Overrides.InstanceType), aws.StringValue(err.LaunchTemplateAndOverrides.Overrides.AvailabilityZone))
+		}
+	}
```

Comment:
Thoughts on encapsulating this inside `updateUnavailableOfferingsCache`. This method is already pretty beefy.
```
for _, err := range createFleetOutput.Errors {
   p.cacheUnavailableOfferings(ctx, err, capacityType)
}
```
or
``` 
if createFleetOutput.Errors != nil {
  p.cacheUnavailableOfferings(ctx, createFleetOutput.Errors, capacityType)
}
```